Extension/src/LanguageServer/configurations.ts

```diff
@@ -2030,6 +2007,43 @@ export class CppProperties {
             const pattern: RegExp = new RegExp(`"[^"]*?(?<="|;)${escapedPath}(?="|;).*?"`, "g");
             const configMatches: string[] | null = curText.match(pattern);
 
+            const expandedPaths: string[] = this.resolveAndSplit([curPath], undefined, this.ExtendedEnvironment, true, false);
+            const incorrectExpandedPaths: string[] = [];
+
+            if (expandedPaths.length > 0) {
+                if (this.rootUri) {
+                    for (const [index, expandedPath] of expandedPaths.entries()) {
+
+                        if (expandedPath.includes("${workspaceFolder}")) {
+                            expandedPaths[index] = this.resolvePath(expandedPath, false);
+                        } else {
+                            expandedPaths[index] = this.resolvePath(expandedPath);
+                        }
+
+                        const checkPathExists: any = util.checkPathExistsSync(expandedPaths[index], this.rootUri.fsPath + path.sep, isWindows, false);
+                        if (!checkPathExists.pathExists) {
+                            // If there are multiple paths, store any non-existing paths to squiggle later on.
+                            incorrectExpandedPaths.push(expandedPaths[index]);
+                        }
+                    }
+                }
+            } else {
```

Comment:
This is easier to read like

```
if (expandedPaths.length === 0) {
    continue;

if (this.rootUri) {
```

i.e. do an early continue. That also makes the block move to the right more.